src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1800,14 +1893,18 @@ def prepare_inputs_for_generation(
         encoder_outputs=None,
         attention_mask=None,
         decoder_attention_mask=None,
+        cache_position=None,
         **kwargs,
     ):
         decoder_position_ids = None
         if decoder_attention_mask is not None:
             decoder_position_ids = (decoder_attention_mask.cumsum(-1) - 1).clamp(min=0)
 
         if past_key_values is not None:
-            past_length = past_key_values[0][0].shape[2]
+            if isinstance(past_key_values[0], Cache):
+                past_length = past_key_values[0].get_seq_length()
```

Comment:
IMO this should not be here, instead if cache position is not None, we should use cache positions!